tests/test_views.py

```diff
@@ -28,406 +28,444 @@
 def test_get_workflows(app, default_user):
     """Test get_workflows view."""
     with app.test_client() as client:
-        with patch(
-            "reana_server.rest.workflows.current_rwc_api_client",
-            make_mock_api_client("reana-workflow-controller")(),
-        ):
-            res = client.get(
-                url_for("workflows.get_workflows"),
-                query_string={"user_id": default_user.id_,
-                              "type": "batch"},
-            )
-            assert res.status_code == 403
-
-            res = client.get(
-                url_for("workflows.get_workflows"),
-                query_string={"access_token": default_user.access_token,
-                              "type": "batch"},
-            )
-            assert res.status_code == 200
+        mock_current_user = Mock(is_authenticated=False)
+        with patch('flask_login.utils._get_user',
+                   Mock(return_value=mock_current_user)):
+            with patch(
+                "reana_server.rest.workflows.current_rwc_api_client",
+                make_mock_api_client("reana-workflow-controller")(),
+            ):
+                res = client.get(
+                    url_for("workflows.get_workflows"),
+                    query_string={"user_id": default_user.id_,
```

Comment:
This was not introduced by you, but since you are changing the code, could you remove the `user_id` parameter? [It is no longer in the API](https://reana-server.readthedocs.io/en/latest/_static/api.html#operation/get_workflows). Thanks!